lib/save.js

```diff
@@ -0,0 +1,18 @@
+var thenify = require('thenify')
+var fs = require('fs')
+var writeFile = thenify(fs.writeFile)
+
+module.exports = function save (pkg, installedPackages, saveType) {
+  var packageJson = pkg.pkg
+  packageJson[saveType] = packageJson[saveType] || {}
+  installedPackages.forEach(function (dependency) {
+    // Assumes package name of in the format of "package-name@x.x.x"
+    // TODO: Support other install sources (Github, scoped pkgs etc.)
```

Comment:
I think a neater way to do this is to make `installMultiple()` work like this:

```js
installMultiple(...)
  .then(packages => {
    // sift through `packages` for ones that match `cli.input`, and save those
  })
```

to do this, `install()` should simply resolve to the necessary package data at the end:

```
function install (...) {
  // these stuff are already available in `install()`.
  var pkg = {
    name: 'lodash',
    fulldata: /* package.json here */,
    ...
  }

  return /* snip */
    .then(_ => pkg) // <-- resolve the Promise to that stuff so that it can be picked up later
}
```